### PR TITLE
[doc] Update the user guide and the development document in Colossal-Inference

### PR DESCRIPTION
**Pull request by KKZ20**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [ ] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

Re-write the user guide for running inference examples and the overall development document of Colossal-Inference.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [ ] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [ ] I have performed a self-review of my code
- [ ] I have added thorough tests.
- [ ] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: examples/inference/README.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Colossal-Inference
+
+
+## Table of Contents
+
+-   📚 Introduction
+
+- 🔨 Installation
+- 🚀 Quick Start
+- 💡Usage
```
Comment (FrankLeeeee) [inline examples/inference/README.md:10]: There is an empty line between introduction and installation.

Comment (FrankLeeeee) [reply examples/inference/README.md:10]: Make them clickable by using something like

- [📚 Introduction](#📚-introduction)

You can test it on your own.

Comment (FrankLeeeee) [reply examples/inference/README.md:10]: Do the same for the development doc as well.

File: examples/inference/README.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+Install Colossal-AI from source:
+
+```shell
+# Clone Colossal-AI repository to your workspace
+git clone https://github.com/hpcaitech/ColossalAI.git
+cd ColossalAI
+
+# Install Colossal-AI from source
+CUDA_EXT=1 pip install .
```
Comment (FrankLeeeee) [inline examples/inference/README.md:49]: You can remove the CUDA_EXT=1 so make installation easier. Any cuda extension will be automatically installed during runtime if needed.

File: examples/inference/README.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+```shell
+cd ColossalAI/inference/dokcer
+docker build
+```
+
+Run the following command to start the docker container in interactive mode.
+
+```shell
+docker run -it --gpus all --name ANY_NAME -v $PWD:/workspace -w /workspace hpcaitech/colossalai-inference:v2 /bin/bash
```
Comment (FrankLeeeee) [inline examples/inference/README.md:85]: Do not use anything like `ANY_NAME`, give a real name.

Comment (FrankLeeeee) [reply examples/inference/README.md:85]: The tag is also weird, the public do not even know we have v1. Use `latest` as our tag.

File: examples/inference/README.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+\[Todo\]: Waiting for Docker file (Li Cuiqing)
+
+## 🚀 Quick Start
+
+You can try the example following the instructions below:
+
+```shell
+cd ColossalAI/examples/inference
+python run_llama_inference.py -p [xxxxx] -b 4 --max_input_len 128 --max_output_len 64 --dtype fp16
```
Comment (FrankLeeeee) [inline examples/inference/README.md:96]: Don't use `[xxxx]` as it is pretty ambiguous, use something like `-p <hf repo name or local path>`. Also, give a runnable example command, e.g. use ColossalAI llama for the `-p` argument.

File: examples/inference/README.md (hunk 3 above)
Comment (KKZ20) [reply examples/inference/README.md:85]: @tiandiao123 please update the tag

File: examples/inference/README.md (hunk 4 above)
Comment (KKZ20) [reply examples/inference/README.md]: Fixed.

File: examples/inference/README.md (hunk 1 above)
Comment (KKZ20) [reply examples/inference/README.md]: Fixed.

File: examples/inference/README.md (hunk 2 above)
Comment (KKZ20) [reply examples/inference/README.md]: Fixed.

**Pull request merged**